server/controllers/stripe.js

```diff
@@ -45,29 +45,51 @@ exports.retrieveCustomer = async (req, res) => {
   res.send(customer);
 };
 
-// @route POST /stripe/payment/create
-// @desc creates new payment method
+// @route GET /stripe/customers/retrieveAll
+// @desc retrieve customer
 // @access Public
-exports.createPaymentMethod = async (req, res) => {
-  const paymentMethod = await stripe.paymentMethods.create({
-    type: "card",
-    card: {
-      number: "4242424242424242",
-      exp_month: 12,
-      exp_year: 2022,
-      cvc: "314",
-    },
-  });
-  res.send(paymentMethod);
+exports.retrieveAllCustomers = async (req, res) => {
+  const customers = await stripe.customers.list();
+  res.send(customers);
 };
 
-// @route GET /stripe/payment/all
+// @route GET /stripe/payment/all/:customerId
 // @desc gets all payment methods
 // @access Public
 exports.getAllPaymentMethods = async (req, res) => {
   const paymentMethods = await stripe.paymentMethods.list({
-    customer: "cus_Kk1rd4KHOzlRC0",
+    customer: req.params.customerId,
```

Comment:
We should restrict the retrieval of payment methods only to the logged in user!

Reply:
I fixed this by using req.user.stripeId to access the logged-in customer's payment method (had to add protect middleware to the "GET /stripe/payment/all/:customerId" route). 

To add stripeId to User, I had to modify the User model (to include stripeId parameter) and "POST /auth/register" route  (to create Stripe customer upon register)